binding/ios/PorcupineManager.swift

```diff
@@ -138,14 +117,15 @@ public class PorcupineManager {
         }
 
         // Only check if it's denied, permission will be automatically asked.
-        let audioSession = AVAudioSession.sharedInstance()
-        if audioSession.recordPermission == .denied {
+        guard try voiceProcessor.hasPermissions() else {
             throw PorcupineManagerError.recordingDenied
         }
         
-        try audioSession.setCategory(AVAudioSession.Category.playAndRecord, options: [.mixWithOthers, .defaultToSpeaker, .allowBluetooth])
-        
-        try audioInputEngine.start()
+        try voiceProcessor.start(
+            frameLength: Porcupine.frameLength,
+            sampleRate: Porcupine.sampleRate,
+            audioCallback: self.audioCallback
```

Comment:
Is there a way you can subscribe/unsubscribe callbacks from the voice processor? That way, with the singleton pattern, you can just add audio callbacks on-the-fly and remove them when you no longer want to be notified.